docs/content/dagster-cloud/developing-testing/branch-deployments/using-branch-deployments-with-github.mdx

```diff
@@ -0,0 +1,302 @@
+
+   <Image
+   alt="Highlighted repository URI in the AWS ECR console"
+   src="/images/dagster-cloud/developing-testing/branch-deployments/aws-ecr-uri.png"
+   width={1086}
+   height={159}
+   />
+
+3. [**Create an IAM user.**](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) This user must:
+
+   - Have push access to the ECR repository, and
+   - Have programmatic access to AWS using an [access key](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html)
+
+   After the user is created, save the **Access key ID** and **Secret access key** values shown on the confirmation page:
+
+   <Image
+   alt="Highlighted repository URI in the AWS ECR console"
+   src="/images/dagster-cloud/developing-testing/branch-deployments/aws-iam-user-keys.png"
+   width={999}
+   height={420}
+   />
+
+</details>
+
+<details>
+	<summary><strong>DOCKER AGENTS</strong></summary>
+
+1. Set up a new Docker agent. Refer to the [Docker agent setup guide](/dagster-cloud/deployment/agents/docker) for instructions.
+2. After the agent is set up, modify the `dagster.yaml` file as follows:
+
+   - Set the `dagster_cloud_api.branch_deployments` field to `true`
+   - Remove any `deployment` field(s)
+
+   For example:
+
+   ```yaml
+   # dagster.yaml
+
+   instance_class:
+     module: dagster_cloud.instance
+     class: DagsterCloudAgentInstance
+
+   dagster_cloud_api:
+     agent_token: <YOUR_AGENT_TOKEN>
+     branch_deployments: true ## true enables branch deployments
+
+   user_code_launcher:
+     module: dagster_cloud.workspace.docker
+     class: DockerUserCodeLauncher
+     config:
+       networks:
+         - dagster_cloud_agent
+       server_ttl:
+         enabled: true
+         ttl_seconds: 7200 #2 hours
+   ```
+
+</details>
+
+<details>
+  <summary><strong>KUBERNETES AGENTS</strong></summary>
+
+1. Set up a new Kubernetes agent. Refer to the [Kubernetes agent setup guide](/dagster-cloud/deployment/agents/Kubernetes/running-configuring-kubernetes-agent) for instructions.
+
+2. After the agent is set up, modify your Helm values file to include the following:
+
+   ```yaml
+   dagsterCloud:
+     branchDeployments: true
+   ---
+   workspace:
+     serverTTL:
+       enabled: true
+       ttlSeconds: 7200
+   ```
+
+</details>
+
+---
+
+## Step 3: GitHub app
+
+TODO?
+
+---
+
+## Step 4: Configure the GitHub repository
+
+In this section, you'll:
+
+1. [Add the Docker registry to `cloud_workspace.yaml`](#step-41-add-the-docker-registry-to-cloud_workspaceyaml)
+2. [Configure GitHub Action secrets](#step-42-configure-github-action-secrets)
+3. [Update GitHub Workflow files](#step-43-update-github-workflow-files)
+4. [TODO: Verify builds]()
+
+### Step 4.1: Add the Docker registry to cloud_workspace.yaml
+
+In the `cloud_workspace.yaml` file, replace `build.registry` with the registry used by the [agent you created in Step 2](#step-2-create-and-configure-an-agent).
```

Comment:
We may be renaming this to `dagster_cloud.yaml`, but should be an easy find+replace if so